### PR TITLE
Initial version.

### PR DESCRIPTION
**Pull request by pwhelan**
'Nuff said.

### CODE REVIEW

**Comment by niedbalski** [comment]
@pwhelan please add a LICENSE (Apache) , also, go.mod/sum (weird it doesn't have a nats.io entry)?

**Review by niedbalski — CHANGES_REQUESTED** [review]
@pwhelan please add a LICENSE (Apache) , also, go.mod/sum (weird it doesn't have a nats.io entry) go mod init/tidy?

**Comment by pwhelan** [comment]
>  go.mod/sum (weird it doesn't have a nats.io entry) go mod init/tidy?

The go.sum never gets generated, even with mod init and tidy. 🤷

**Review by niedbalski — CHANGES_REQUESTED** [review]
Also, package name can be just transport.

**Comment by niedbalski** [comment]
> > go.mod/sum (weird it doesn't have a nats.io entry) go mod init/tidy?
> 
> The go.sum never gets generated, even with mod init and tidy. 🤷

go get -u github.com/nats-io/nats.go 
go mod tidy

File: transport.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
+package core_transport
```
Comment (niedbalski) [inline transport.go]: underscore in package names aren't really used in go (https://go.dev/blog/package-names). Let's just called transport.

**Review by niedbalski — CHANGES_REQUESTED** [review]

**Review by nicolasparada — APPROVED** [review]

**Review by niedbalski — APPROVED** [review]

**Pull request merged**